make/autoconf/flags-ldflags.m4

```diff
@@ -71,9 +71,6 @@ AC_DEFUN([FLAGS_SETUP_LDFLAGS_HELPER],
       LIBJSIG_HASHSTYLE_LDFLAGS="-Wl,--hash-style=both"
     fi
 
-    # Add -z defs, to forbid undefined symbols in object files.
-    BASIC_LDFLAGS="$BASIC_LDFLAGS -Wl,-z,defs"
```

Comment:
Removing this seems like it just papers over the problem and would also allow for undefined symbols to creep in.  The commit says the missing symbol was 'environ', can you provide more context?  Was that in linking the VM?

Reply:
I'll post the log tomorrow

Reply:
first error is (missing lpthread): https://gist.github.com/MikaelUrankar/6a5ca41b1a20ab1a4a7c2d9cb8b27903

second error is:
https://gist.github.com/MikaelUrankar/ce3e94e2c7378950a7d00a570ad37246

Reply:
I added some comments on the log entries at the links above.

Reply:
Link with -pthread
Use -Wl,-z,defs on !bsd platform